bin/bb8/settings.py

```diff
@@ -21,8 +22,9 @@ def __init__(self, path=config_path):
         self.starport = config["starport"]
         self.targets = list(Settings.parse_target(t) for t in config["targets"])
 
-        if "instance_guid" in config:
-            self.instance_guid = config["instance_guid"]
+        if isfile(machine_id_path):
+            with open(machine_id_path, 'r') as f:
+                self.instance_guid = f.read().replace('\n', '')
```

Comment:
Maybe it should just error if there's no machine id now? What's the reason for falling back to null?

Reply:
This is because `settings.py` is also used during container build, when there is no such file (because it only exists when mounted in.) Could just not set the property at all I guess? But probably better to be explicit?

Reply:
(which is the same reason the if else check exists in the current code)